ibis/expr/operations/relations.py

```diff
@@ -169,6 +169,8 @@ def _clean_join_predicates(left, right, predicates):
             pred = lk == rk
         elif isinstance(pred, str):
             pred = left.to_expr()[pred] == right.to_expr()[pred]
+        elif pred is True or pred is False:
```

Comment:
```suggestion
        elif isinstance(pred, bool):
```

Reply:
It better aligns with the rest of the conditions and also support bool subclasses.

Reply:
`bool` cannot be subclassed.

```
In [1]: class MyBool(bool):
   ...:     pass
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 class MyBool(bool):
      2     pass

TypeError: type 'bool' is not an acceptable base type
```